internal/controllers/gateway/route_utils.go

```diff
@@ -233,6 +244,11 @@ func getSupportedGatewayForRoute[T types.RouteT](ctx context.Context, mgrc clien
 				// If there is no matchingHostname, the gateway Status Condition Accepted
 				// must be set to False with reason NoMatchingListenerHostname
 				reason = gatewayv1beta1.RouteReasonNoMatchingListenerHostname
+			} else if (parentRef.SectionName) != nil && !allowedByListenerName {
+				// If ParentRef specified listener names but none of the listeners matches the name,
+				// the gateway Status Condition Accepted must be set to False with reason RouteReasonNoMatchingParent.
+				// TODO: which listenerName to fill in for such case?
```

Comment:
I believe the current does the correct thing (although I'm not 100% sure :) ). The status in `HTTPRoute` should contain the statuses of `ParentRef`'s, not the gateway listener's which we did not match.

Reply:
removed the `// TODO`.